linalg/densemat.cpp

```diff
@@ -3971,7 +4053,12 @@ void LUFactors::Factor(int m)
             }
          }
       }
-      MFEM_ASSERT(data[i+i*m] != 0.0, "division by zero");
+
+      if ( abs(data[i+i*m]-0.0) <= 1.e-9 )
```

Comment:
Not sure about the implications of this magic number

Reply:
This was done instead of `if ( data[i+i*m] == 0.0 )`. What would you suggest? 

I agree magic numbers are generally bad. We could have an optional argument to the `LUFactors::Factor(int m, double TOL=1.e-9)` that way the application can set it. Is this preferable?

Reply:
I added an optional tolerance argument to the `LUFactors::Factor` method. This avoids the magical numerical value in this check. The default is still `1.e-9`, but the caller may change this. Let me know what you think.